src/components/File.jsx

```diff
@@ -94,6 +98,20 @@ class File extends Component {
     )
   }
 
+  renderFilenameInput (attributes) {
```

Comment:
I think this shall be a component taking `isRenaming`, `attributes`, `opening`, `selectionModeActive` and returning either a sub component called `RenameInput` or `FileCell`. It worth be clearer and more homogeneous.